src/livox_laser_simulation/livox_points_plugin.cpp

```diff
@@ -0,0 +1,370 @@
+//
+// Created by lfc on 2021/2/28.
+//
+
+#include "livox_laser_simulation/livox_points_plugin.h"
+#include <ros/ros.h>
+#include <sensor_msgs/PointCloud.h>
+#include <sensor_msgs/PointCloud2.h>
+#include <sensor_msgs/point_cloud_conversion.h>
+#include <gazebo/physics/Model.hh>
+#include <gazebo/physics/MultiRayShape.hh>
+#include <gazebo/physics/PhysicsEngine.hh>
+#include <gazebo/physics/World.hh>
+#include <gazebo/sensors/RaySensor.hh>
+#include <gazebo/transport/Node.hh>
+#include "livox_laser_simulation/csv_reader.hpp"
+#include "livox_laser_simulation/livox_ode_multiray_shape.h"
+#include <iostream>
+#include <filesystem>
+
+namespace gazebo {
+
+GZ_REGISTER_SENSOR_PLUGIN(LivoxPointsPlugin)
+
+LivoxPointsPlugin::LivoxPointsPlugin() {}
+
+LivoxPointsPlugin::~LivoxPointsPlugin() {}
+
+void convertDataToRotateInfo(const std::vector<std::vector<double>> &datas, std::vector<AviaRotateInfo> &avia_infos) {
+    avia_infos.reserve(datas.size());
+    double deg_2_rad = M_PI / 180.0;
+    for (auto &data : datas) {
+        if (data.size() == 3) {
+            avia_infos.emplace_back();
+            avia_infos.back().time = data[0];
+            avia_infos.back().azimuth = data[1] * deg_2_rad;
+            avia_infos.back().zenith = data[2] * deg_2_rad - M_PI_2;  //转化成标准的右手系角度
```

Comment:
I take it that this code has been lifted from third party software?

Reply:
Yup, the lib lives in https://github.com/Livox-SDK/livox_laser_simulation